compose/config.go

```diff
@@ -12,6 +12,9 @@ type Config struct {
 	// IDTokenLifespan sets how long an id token is going to be valid. Defaults to one hour.
 	IDTokenLifespan time.Duration
 
+	// RefreshTokenLifespan sets how long a refresh token is going to be valid. Defaults to zero meaning no expiry. Negative values mean the token is permanent.
```

Comment:
how is no expiry different than permanent? No expiry expires with the first use?

Reply:
Essentially. The implementation of the system was to have a refresh token that is one time use, but has no expiration. Adding a lifespan, defaulting it to the old implementation, seemed more reasonable rather than a `bool` that says it's permanent. This allows for the old usage to remain the same, adds the ability be be permanent and allows the token to expire.